dvc/api.py

```diff
@@ -1,15 +1,67 @@
+from contextlib import contextmanager
+import os
+import tempfile
+
+try:
+    from contextlib import _GeneratorContextManager as GCM
+except ImportError:
+    from contextlib import GeneratorContextManager as GCM
+
+from dvc.utils import remove
+from dvc.utils.compat import urlparse
 from dvc.repo import Repo
+from dvc.external_repo import ExternalRepo
 
 
-def get_url(path, repo_dir=None, remote=None):
-    """Returns an url of `path` in default or specified remote"""
-    repo = Repo(repo_dir)
-    out, = repo.find_outs_by_path(path)
-    remote_obj = repo.cloud.get_remote(remote)
-    return str(remote_obj.checksum_to_path_info(out.checksum))
+def get_url(path, repo=None, remote=None):
+    """Returns an url of a resource specified by path in repo"""
+    with _make_repo(repo) as _repo:
+        abspath = os.path.join(_repo.root_dir, path)
+        out, = _repo.find_outs_by_path(abspath)
+        remote_obj = _repo.cloud.get_remote(remote)
+        return str(remote_obj.checksum_to_path_info(out.checksum))
 
 
-def open(path, repo_dir=None, remote=None, mode="r", encoding=None):
+def open(*args, **kwargs):
```

Comment:
I agree with @shcheklein . We should either list the args here or at least create a proper docstring that has arguments listed.